lib/class-wp-rest-customizer-nonces.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * WP_Rest_Customizer_Nonces class.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Class that returns the customizer "save" nonce that's required for the
+ * batch save operation using the customizer API endpoint.
+ */
+class WP_Rest_Customizer_Nonces extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'customizer-nonces';
+	}
+
+	/**
+	 * Registers the necessary REST API routes.
+	 *
+	 * @access public
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/get-save-nonce',
+			array(
+				array(
+					'methods'  => WP_REST_Server::READABLE,
+					'callback' => array( $this, 'get_save_nonce' ),
```

Comment:
Can we add a permission check to this?

Reply:
We could, but actually let me turn this question on you - does it make sense to add one? This is only returning a nonce to be sent to `admin-ajax.php` alongside updated navigation structure, and `admin-ajax.php` does it's own permission checks.

Reply:
Is it 100% required and doing so will definitely lead to a security issue? Probably not. But why increase the attack surface?

Reply:
I just added `permission_callback` 👍

Reply:
👍 thanks lgtm!